CalPatRec/src/PhiClusterFinder_module.cc

```diff
@@ -0,0 +1,566 @@
+    void fillTimeSpectrum(int ClustNo, const std::vector<StrawHitIndex>& ordchcol);
+    void fillTimeSpectrumAlt(const std::vector<StrawHitIndex>& ordchcol);
+    void findPeaks  (TimeCluster& tc, const std::vector<StrawHitIndex>& ordchcol);
+    void assignHits(TimeCluster& otc, const std::vector<StrawHitIndex>& ordchcol);
+    void initCluster(TimeCluster& tc);
+    float checkdelta(TimeCluster& tc, int ClustNo, const std::vector<StrawHitIndex>& ordchcol);
+    void addCaloClusters(TimeClusterCollection& tccolnew, const TimeClusterCollection& tccol);
+};
+
+ PhiClusterFinder::PhiClusterFinder(const art::EDProducer::Table<Config>& config):
+    art::EDProducer{config},
+    _diag         (config().diag()),
+    _debug        (config().debug()),
+    _minnsh       (config().minNSH()),
+    _threshold    (config().threshold()),
+    _nphiclusters (config().nPhiClusters()),
+    _phimin       (config().phiMin()),
+    _phimax       (config().phiMax()),
+    _nphibins     (config().nPhiBins()),
+    _mindphi      (config().minDPhi()),
+    _pitch        (config().pitch()),
+    _tmin         (config().tMin()),
+    _tmax         (config().tMax()),
+    _tbin         (config().tBin()),
+    _t0calc       (config().t0Calc()),
+    _ymin         (config().yMin()),
+    _maxdt        (config().maxDt()),
+    _minsigma     (config().minSigma()),
+    _npeak        (config().peakWidth()),
+    _chToken      {consumes<ComboHitCollection>(config().ComboHitCollection()) },
+    _tcToken      {consumes<TimeClusterCollection>(config().TimeClusterCollection()) },
+    _useCC        (config().useCC())
+    {
+      art::ServiceHandle<art::TFileService> tfs;
+      _hist1 = tfs->make<TH1F>( "hist1" , "phi spectrum",_nphibins,_phimin,_phimax);
+      unsigned nbinst = (unsigned)rint((_tmax-_tmin)/_tbin);
+      _timespec = TH1F("timespec","time spectrum",nbinst,_tmin,_tmax);
+      produces<TimeClusterCollection>();
+
+      if (_diag != 0) _hmanager = art::make_tool<ModuleHistToolBase>(config().DiagPlugin," ");
+      else            _hmanager = std::make_unique<ModuleHistToolBase>();
+    }
+
+  void PhiClusterFinder::beginJob() {
+    if (_diag > 0) {
+      art::ServiceHandle<art::TFileService> tfs;
+      _hmanager->bookHistograms(tfs);
+    }
+  }
+
+  void PhiClusterFinder::produce(art::Event & event ){
+    _iev = event.id().event();
+    std::unique_ptr<TimeClusterCollection> tccol1(new TimeClusterCollection);
+    auto const& chH = event.getValidHandle(_chToken);
+    _chcol = chH.product();
+
+    // Save the sum phi of a cluster to calculate delta phi between the clusters
+    _phitotal.clear();
+    _nhitotal.clear();
+    _phitotal.resize(_nphiclusters);
+    _nhitotal.resize(_nphiclusters);
+
+    for(int m=0;m<_nphiclusters;m++){
+      _phitotal[m] = 0.;
+      _nhitotal[m] = 0.;
+    }
+    auto const& tcH = event.getValidHandle(_tcToken);
+    const TimeClusterCollection& tccol(*tcH);
```

Comment:
this variable is not necessary if you use `_data._tccol` or alternatively, delete `_data._tccol` as it is used nowhere in the module.

Reply:
`_data._tcol` is used for the diagnostics so that one can check the no. of input time clusters and compare with the no. of output phi clusters. I've changed it to `if(_diag > 0)` then initialize `_data._tccol`.